### PR TITLE
fix a bunch of bugs in qml window placement

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by ZappoMan** [comment]
Fixes these bugs:
- https://app.asana.com/0/26320649018758/116220905173221
- https://app.asana.com/0/26320649018758/115221114615741
- https://app.asana.com/0/26320649018758/115246934812899
- https://app.asana.com/0/26320649018758/115748486008826
- https://app.asana.com/0/26320649018758/115246934812896

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ab7b660d44e3e6095a19d583575b56c3738d998 available at: https://gist.github.com/6ae6b400f576fab036129f8d92148e7a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ab7b660d44e3e6095a19d583575b56c3738d998 available at: https://gist.github.com/6ae6b400f576fab036129f8d92148e7a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 716d9e554c90802d622a7ae3f70e309fbb7d9c57 available at: https://gist.github.com/6ae6b400f576fab036129f8d92148e7a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.dmg

**Comment by judasshuffle** [comment]
smoke test
    reset your settings and make sure that the "help pupup" is centered in the window
    move the mute/audio meter control off the window so it's outside the main display area, resize the window and see that it pops back to be visible ** That worked **
    after reset settings make sure that new dialogs like the settings dialog pop up in center view
    using the HMD move an overlay window so that it's "off of the preferred viewing area" - switch to desktop and verify that the window is now in view on the desktop
**that worked 2 yay 2 fer 2
**
Pass:)

File: interface/resources/qml/desktop/Desktop.qml [hunk 1]
```diff
@@ -279,13 +285,58 @@ FocusScope {
...
+
+        // If we've noticed that our recommended desktop rect has changed, record that change here.
+        if (recommendedRect != newRecommendedRect) {
+            recommendedRect = newRecommendedRect;
+        }
+
+    }
+
+    function repositionOnVisible(item) {
+        console.warn("repositionOnVisible() item:" + item);
```
Comment (jherico) [inline interface/resources/qml/desktop/Desktop.qml]: remove debug logging

File: interface/resources/qml/desktop/Desktop.qml [hunk 2]
```diff
@@ -279,13 +285,58 @@ FocusScope {
...
+        var targetWindow = d.getDesktopWindow(item);
+        console.warn("repositionOnVisible() targetWindow:" + targetWindow);
```
Comment (jherico) [inline interface/resources/qml/desktop/Desktop.qml]: remove debug ogging

File: interface/resources/qml/windows-uit/Window.qml [hunk 3]
```diff
@@ -31,7 +31,7 @@ Fadable {
     // decorations can extend outside it.
     implicitHeight: content ? content.height : 0
     implicitWidth: content ? content.width : 0
-    x: -1; y: -1
+    x: -9999; y: -9999
```
Comment (jherico) [inline interface/resources/qml/windows-uit/Window.qml]: make magic numbers constants.  Either in a local `QtObject { id: d }` or just at the top level as a `readonly int INVALID_POSITION: -9999`

Comment (ZappoMan) [inline interface/resources/qml/windows-uit/Window.qml]: How do I do this in a way that allows me to use the same constant in both places?

**Comment by jherico** [comment]
adding DNM while the invalid QML is  in place.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e7d26a8a63102d5fd3b9ddcb997efc337b3bdf6d available at: https://gist.github.com/6ae6b400f576fab036129f8d92148e7a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 82f905dc8439972106d263070425ee9ce3a5ed30 available at: https://gist.github.com/6ae6b400f576fab036129f8d92148e7a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7762.dmg

File: interface/resources/qml/windows/Window.qml [hunk 4]
```diff
@@ -19,7 +19,7 @@ Fadable {
     // decorations can extend outside it.
     implicitHeight: content ? content.height : 0
     implicitWidth: content ? content.width : 0
-    x: -1; y: -1
```
Comment (howard-stearns) [inline interface/resources/qml/windows/Window.qml]: I thought qml/windows/ was supposed to remain unchanged from qt, with our changes going into qml/windows-uit/, no?

Comment (ZappoMan) [inline interface/resources/qml/windows/Window.qml]: No -- those are just the old styled windows... we still have some of those, like the web browser, and others... they aren't specific to Qt... but hopefully our use of them will go away eventually.

Basically, to honor your spec, these needed to change too.

Comment (howard-stearns) [inline interface/resources/qml/windows/Window.qml]: ok, cool.

**Pull request merged**